Bio/SeqIO/QualityIO.py

```diff
@@ -1867,18 +1899,23 @@ def as_fastq_illumina(record):
             "Record %s has sequence length %i but %i quality scores"
             % (record.id, len(seq_str), len(qualities_str))
         )
-    id = _clean(record.id)
+    id_ = _clean(record.id) if record.id else ""
     description = _clean(record.description)
-    if description and description.split(None, 1)[0] == id:
+    if description and description.split(None, 1)[0] == id_:
         title = description
     elif description:
-        title = "%s %s" % (id, description)
+        title = "%s %s" % (id_, description)
     else:
-        title = id
+        title = id_
     return "@%s\n%s\n+\n%s\n" % (title, seq_str, qualities_str)
 
 
-def PairedFastaQualIterator(fasta_source, qual_source, alphabet=None, title2ids=None):
+def PairedFastaQualIterator(
+    fasta_source: TextIO,
+    qual_source: TextIO,
+    alphabet=None,
+    title2ids: Optional[Any] = None,
```

Comment:
This is another place where the hint of `title2ids` can be made more precise (see comment regarding `title2ids` alias above).